src/compiler/checker.ts

```diff
@@ -2452,6 +2452,69 @@ module ts {
             return result;
         }
 
+        function getBaseTypes(type: InterfaceType): ObjectType[] {
+            if (!(<InterfaceTypeWithBaseTypes>type).baseTypes) {

```

Comment:
we have ObjectType and ResolvedObjectType.  Should we have InterfaceType and ResolvedInterfaceType?